app/models/portfolio.rb

```diff
@@ -11,14 +11,23 @@
 
 
 class Portfolio < ApplicationRecord
+  include Discard::Model
   acts_as_tenant(:tenant)
+  default_scope -> { kept }
 
-  validates :name, presence: true, :uniqueness => { :scope => :tenant_id }
+  validates :name, :presence => true, :uniqueness => { :scope => %i(tenant_id discarded_at) }
   validates :image_url, :format => { :with => URI::DEFAULT_PARSER.make_regexp }, :allow_blank => true
   validates :enabled_before_type_cast, :format => { :with => /\A(true|false)\z/i }, :allow_blank => true
 
   has_many :portfolio_items, :dependent => :destroy
 
+  before_discard do
+    if portfolio_items.map(&:discard).any? { |result| result == false }
+      Rails.logger.error("Failed to discard items from Portfolio #{id} - not discarding portfolio")
```

Comment:
@lindgrenj6 Few things.  I think we should not stop the transaction, but make sure the `errors` object is populated (object or instance variable, I'm not sure I remember ) with the failures, and continue on leaving only those we couldn't discard ( in reference to what @gmcculloug said earlier. )

I also would like to add the error handling at the controller side, so the `errors.messages` ( or `errors.full_messages` )  are sent back to the caller via the `render :json` method

Also we need to make sure we return false if there is an error so the transaction not allow the `Portfolio` to be deleted if there are errors.

Reply:
Sorry - I see you pulled out the `throw(:abort)` and are using false here now.

Reply:
Ok, got this refactored a bit to add in error messages for every portfolio item. It also returns a `422: Unprocessable Entity` when discard-ation fails.

Reply:
Small nit, can we also include the name in the `Rails.logger.error` string?

Reply:
Yes definitely - should have had it there in the first place.